Cpp_test/lib/file-handler.h

```diff
@@ -0,0 +1,33 @@
+#ifndef FILE_HANDLER_H_
+#define FILE_HANDLER_H_
+
+#include <iostream>
+#include <fstream>
+#include <utility>
+#include <boost/filesystem.hpp>
+
+enum class FileMode
+{
+	NONE, READ, WRITE
+};
+
+class FileHandler
+{
+	protected:
+		std::fstream fs{};
+		char *file_name;
+		FileMode mode{FileMode::NONE};
+	public:
+		explicit FileHandler(char *fn, FileMode m) 
+			: file_name(std::move(fn)), mode(m)
```

Comment:
I don't understand the objective of using std::move here. Can you explain?

Reply:
I use it to avoid copying, so I can move the value of fn and original value is no longer available.

Reply:
But char* is a pointer, and avoiding a copy of a pointer is not really relevant.

And std::move(pointer) == make a copy of the pointer.


Basically, std::move is only useful if the object implements the move method.

Reply:
ah, yes. You are right, I will modify the code again.